helix-gateway/src/main/proto/HelixGatewayService.proto

```diff
@@ -0,0 +1,52 @@
+syntax = "proto3";
+
+package proto.org.apache.helix.gateway;
+
+message SingleTransitionMessage {
+  enum TransitionType {
+    ADD_SHARD = 0;
+    DELETE_SHARD = 1;
+    CHANGE_ROLE = 2;
+  }
+  string transitionID = 1;    // ID of transition message
+  TransitionType transitionType = 2;   // Transition type for shard operations
+  string resourceID = 3;              // Resource ID
+  string shardID = 4;                 // Shard to perform operation
+  string startState = 5;      // Shard start state
+  string targetState = 6;     // Shard target state.
+}
+
+message StateTransitionMessage{
```

Comment:
We call single transition message previously, then following naming should be consistent. This should be called as TransitionMessages as well.